### PR TITLE
chore: unify external logger code

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

dublicated code for external datadog logger

### Solutions

move the code to its own source directory and add it to internal app flavours

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

build the app and datadog is there

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [5291](https://10.10.124.134/job/AR-build-pipeline/5291/) **failed**.

**Comment by AndroidBob** [comment]
Build [5292](https://10.10.124.134/job/AR-build-pipeline/5292/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0-9939278-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0-9939278-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5313](https://10.10.124.134/job/AR-build-pipeline/5313/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0-9953140-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0-9953140-staging-compat.apk)

File: app/build.gradle.kts [hunk 1]
```diff
@@ -60,6 +61,22 @@ private fun getFlavorsSettings(): NormalizedFlavorSettings =
     }
 
 android {
+    defaultConfig {
+
+        val localProperties = Properties()
+        val localPropertiesFile = rootProject.file("local.properties")
+        if (localPropertiesFile.exists()) {
+            localProperties.load(localPropertiesFile.inputStream())
+        }
```
Comment (yamilmedina) [inline app/build.gradle.kts]: Suggestion: We can make use of this handy function read from `local.properties` :D 

https://github.com/wireapp/wire-android/blob/b47aaae9f8f58a4bec19d88097e30d7630be553a/buildSrc/src/main/kotlin/ProjectExtensions.kt#L30

**Review by yamilmedina — COMMENTED** [review]
Looking fantastic, I added a suggestion about getting the local props and go for reusage 💪

**Review by yamilmedina — APPROVED** [review]
👏 🐇

**Review by vitorhugods — COMMENTED** [review]
Instead of `external-logger-enabled`, what about `public` vs `private` source sets?

`prod` and `fdroid` can use `public` (no external logger). The other flavours can use `private` (with external logger).

I'd like to avoid setting a pattern where every different configurable feature will will have its own source set.

I'd also take the opportunity to remove the `internal` flavour 😅

**Comment by MohamadJaara** [comment]
using private vs public is a nice idea i will give it a try, but for Internal i have another plan to turn it into an edge environment (publish with every push to RC)
@vitorhugods

**Comment by AndroidBob** [comment]
Build [5318](https://10.10.124.134/job/AR-build-pipeline/5318/) **failed**.

**Comment by vitorhugods** [comment]
> using private vs public is a nice idea i will give it a try, but for Internal i have another plan to turn it into an edge environment (publish with every push to RC) @vitorhugods

That's a nice idea for `internal`!

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**